lxd/instances_post.go

```diff
@@ -37,6 +37,45 @@ import (
 	"github.com/lxc/lxd/shared/osarch"
 )
 
+func ensureDownloadedImageFitWithinBudget(s *state.State, r *http.Request, op *operations.Operation, p api.Project, img *api.Image, imgAlias string, source api.InstanceSource, imgType string) (*api.Image, error) {
```

Comment:
This should be split into separate commit(s) that are not tightly coupled to instance rebuild changes.